Refactor USAGE.md #1371 #1372 (#1496)
cncf/cnf-testsuite: RATIONALE.md

#### *Test if the CNF crashes when pod delete occurs*: [pod_delete](USAGE.md#heavy_check_mark-test-if-the-cnf-crashes-when-pod-delete-occurs)

> In a distributed system like Kubernetes, likely, your application replicas may not be sufficient to manage the traffic (indicated by SLIs) when some of the replicas are unavailable due to any failure (can be system or application) the application needs to meet the SLO(service level objectives) for this. It's imperative that your application has defences aginst this sort of failure to ensure that the application always has a minimum number of available replicas. 

#### *Test if the CNF crashes when pod memory hog occurs*: [pod_memory_hog](USAGE.md#heavy_check_mark-test-if-the-cnf-crashes-when-pod-memory-hog-occurs)

> > If the memory policies for a CNF are not set and fine grained enough, containers on the node can be killed based on their oom_score and the QoS class a given pod belongs to (bestEffort ones are first to be targeted). This eval is extended to all pods running on the node - thereby causing a bigger blast radius. 

Reviewer: ```suggestion
> If the memory policies for a CNF are not set and granular, containers on the node can be killed based on their oom_score and the QoS class a given pod belongs to (best-effort ones are first to be targeted). This eval is extended to all pods running on the node, thereby causing a bigger blast radius. 
```

such as those that will eventually cause a failure.

#### *To check if logs and data are being routed through fluentd*: [routed_logs](USAGE.md#heavy_check_mark-to-check-if-logs-and-data-are-being-routed-through-fluentd)
> A CNF should have logs managed by a [unified logging layer](https://www.fluentd.org/why) It's considered a best-practice for CNFs to route logs and data through programs like fluentd to analyze and better understand data.

#### *To check if OpenMetrics is being used and or compatible.*: [open_metrics](USAGE.md#heavy_check_mark-to-check-if-open-metrics-is-being-used-and-or-compatible)

> > OpenMetrics is the de-facto standard for transmitting cloud-native metrics at scale, with support for both text representation and Protocol Buffers and brings it into an Internet Engineering Task Force (IETF) standard. A CNF should expose metrics that are [OpenMetrics compatible](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md)

Reviewer: ```suggestion
> OpenMetrics is the de facto standard for transmitting cloud native metrics at scale, with support for both text representation and Protocol Buffers and brings it into an Internet Engineering Task Force (IETF) standard. A CNF should expose metrics that are [OpenMetrics compatible](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md)
```

> CPU and memory [resources should have a limit](https://hub.armo.cloud/docs/c-0009) set for every container or a namespace to prevent resource exhaustion. This control identifies all the Pods without resource limit definitions by checking thier yaml definition file as well as their namespace LimitRange objects. It is also recommended to use ResourceQuota object to restrict overall namespace resources, but this is not verified by this control.

#### *To check if containers have immutable file systems*: [immutable_file_systems](USAGE.md#heavy_check_mark-to-check-if-containers-have-immutable-file-systems)

> > Mutable container filesystem can be abused to gain malicious code and data injection into containers. By default, containers are permitted mostly unrestricted execution within their own context. An attacker who has access to a container, [can create files](https://hub.armo.cloud/docs/c-0017) and download scripts as he wishes, and modify the underlying application running on the container.

Reviewer: ```suggestion
> Mutable container filesystem can be abused to gain malicious code and data injection into containers. By default, containers are permitted unrestricted execution within their own context. An attacker who has access to a container, [can create files](https://hub.armo.cloud/docs/c-0017) and download scripts as they wish, and modify the underlying application running on the container.
```

> Using node ports ties the CNF to a specific node and therefore makes the CNF less
portable and scalable

#### *To test if the recommended labels are being used to describe resources*: [required_labels](USAGE.md#required-labels)

> > Defining and using labels help to identify semantic attributes of your application or Deployment. A common set of labels allows tools to work collaboratively, describing objects in a common manner that all tools can understand. You should use recommended labels describe applications in a way that can be queried.

Reviewer: ```suggestion
> Defining and using labels help identify semantic attributes of your application or Deployment. A common set of labels allows tools to work collaboratively, while describing objects in a common manner that all tools can understand. You should use recommended labels to describe applications in a way that can be queried.
```

#### *To check if the plateform has a default Cluster admin role*: [cluster-admin](USAGE.md#cluster-admin)
> *Role-based access control (RBAC) is a key security feature in Kubernetes. RBAC can restrict the allowed actions of the various identities in the cluster. Cluster-admin is a built-in high privileged role in Kubernetes. Attackers who have permissions to create bindings and cluster-bindings in the cluster can create a binding to the cluster-admin ClusterRole or to other high privileges roles. As a best practice, a principle of least privilege should be followed and cluster-admin privilege should only be used on an as-needed basis.

#### *Check if the plateform is using insecure ports for the API server*: [Control_plane_hardening](USAGE.md#control-plane-harding)
> *The control plane is the core of Kubernetes and gives users the ability to view containers, schedule new Pods, read Secrets, and execute commands in the cluster. Therefore, it should be protected. It is recommended to avoid control plane exposure to the Internet or to an untrusted network and require TLS encryption.

#### *Check if the Dashboard is exposed externally*: [Dashboard exposed](USAGE.md#dashboard-exposed)

> > * If Kubernetes dashboard is exposed externally in Dashboard versions before 2.01, it will allow unauthenticated remote management of the cluster. It's best practive not expose the K8s Dashboard, or any management planes if they're unsecured.

Reviewer: ```suggestion
> * If Kubernetes dashboard is exposed externally in Dashboard versions before 2.01, it will allow unauthenticated remote management of the cluster. It's best practice to not expose the K8s Dashboard or any management planes if they're unsecured.
```
